### PR TITLE
Rubens 2022 12 17

### PR DESCRIPTION
**Pull request by rubenssvn**
Hello,

All programs were written in Java and I've versioned the .jar in case you need to run without compiling.
Every step folder contain a RUNME.md.

I hope you like it :)

Kind regards,
Rubens

### CODE REVIEW

File: step3/src/test/java/rubens/step3/BaconsLawTest.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+	@Test
+	public void shouldThrowExceptionWhenActor1EqualsActor2() {
+		String[] args = {"Kevin Bacon,Kevin Bacon"};
+		assertThrows(IllegalArgumentException.class, () -> baconsLaw.calculateDegrees(args));
+	}
+	
+	@Test
+	public void shouldReturnDegree1BetweenKevinBaconAndTomCruise() throws IOException {
+		String[] args = {"Tom Cruise"};
+		assertEquals(new Integer(1), baconsLaw.calculateDegrees(args));
```
Comment (rubenssvn) [inline step3/src/test/java/rubens/step3/BaconsLawTest.java:37]: > # There are 2 degrees of separation between Tom Cruise and Kevin Bacon
> # Ralph Macchio starred with Tom Cruise in The Outsiders
> # John Lithgow starred with Ralph Macchio in Distant Thunders
> # Kevin Bacon starred with John Lithgow in Footloose.

Here I found degree separation = 1 between Tom Cruise and Kevin Bacon
Tom Cruise starred with Chris Penn in All the Right Moves
Chriss Penn starred with Kevin Bacon in Footloose

Comment (rubenssvn) [inline step3/src/test/java/rubens/step3/BaconsLawTest.java:38]: _There are 2 degrees of separation between Tom Cruise and Kevin Bacon
Ralph Macchio starred with Tom Cruise in The Outsiders
John Lithgow starred with Ralph Macchio in Distant Thunders
Kevin Bacon starred with John Lithgow in Footloose._

Here I found degree separation = 1 between Tom Cruise and Kevin Bacon
Tom Cruise starred with Chris Penn in All the Right Moves
Chriss Penn starred with Kevin Bacon in Footloose

**Comment by rubenssvn** [comment]
Guys, please keep in mind that the output generated on step2 is not automatically loaded to step3. Although I used the same file, I copied manually. I can change the implementation if you want!

Thanks

**Comment by TarabullNinja** [comment]
@rubenssvn Thank you for your submission!

**Pull request closed without merging**